builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -190,7 +208,24 @@ func resourceArmVirtualMachine() *schema.Resource {
 
 						"vhd_uri": {
 							Type:     schema.TypeString,
-							Required: true,
+							Optional: true,
```

Comment:
Hey @brandontosch

Thanks for pushing those updates - sorry for the delay in getting back to you here.

> The issue was that it tries to lookup the conflicts from the root of the schema using the keys provided inside ConflictsWith. So trying to locate storage_data_disk.vhd_uri would never succeed since it doesn't contain the index and storage_data_disk is a list. This would cause validation to always succeed and lead to an error when trying to create the resource.

Ah, apologies - I wasn't aware `ConflictsWith` didn't support this.

> IMHO it makes sense to expect storage_data_disk.vhd_uri to work for ConflictsWith so I added some code during the validation to check the key that is being validated for the index (since it is there) and then add it to the conflicting key being searched for so that it can check for a value appropriately. Added some tests in there too for various scenarios to make sure it doesn't garble anything up for other cases.

I'd agree that the behaviour is expected, so thanks for adding support for this! Given that's a change to the Core it'd need further discussion, so that we're not blocking this PR on that (as it otherwise looks good to merge!) - can we pull the `ConflictsWith` changes out into a separate PR?

Thanks!